src/Stratis.Bitcoin/P2P/PeerAddress.cs

```diff
@@ -0,0 +1,277 @@
+﻿using System;
+using System.Net;
+using NBitcoin.Protocol;
+using Newtonsoft.Json;
+
+namespace Stratis.Bitcoin.P2P
+{
+    [JsonObject]
+    public sealed class PeerAddress
+    {
+        private const int PEER_MAX_LAST_SEEN = 30;
+        private const int PEER_MIN_FAIL_DAYS = 7;
+        private const int PEER_MAX_WEEKLY_ATTEMPTS = 10;
+        private const int PEER_MAX_CONNECTION_RETRIES = 3;
+
+        #region Address Data
+
+        [JsonProperty]
+        private string address;
```

Comment:
This should probably be an IPAddress class (you can create json converters see wallet)
You can actually just create a converted for IPEndpoint and just keep that (instead of a string and a int for a port)

Reply:
Ah awesome, yes I love converters, forgot I can do that!

Reply:
> I love converters 

😄